integration_test/home_widget_test.dart

```diff
@@ -0,0 +1,68 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:get/get.dart';
+import 'package:integration_test/integration_test.dart';
+import 'package:survey/main.dart';
+import 'package:survey/managers/user_manager.dart';
+import 'package:survey/use_cases/get_login_state_use_case.dart';
+import 'package:survey/use_cases/get_survey_detail_use_case.dart';
+import 'package:survey/use_cases/get_surveys_use_case.dart';
+
+import 'fake/fake_get_login_state.dart';
+import 'fake/fake_get_survey_detail_use_case.dart';
+import 'fake/fake_survey_use_case.dart';
+import 'fake/fake_user_manager.dart';
+
+main() {
+  Get.testMode = true;
+
+  group('Validate logged in scenario in Home screen', () {
+    final binding = IntegrationTestWidgetsFlutterBinding.ensureInitialized()
+        as IntegrationTestWidgetsFlutterBinding;
+
+    binding.framePolicy = LiveTestWidgetsFlutterBindingFramePolicy.fullyLive;
+    Get.put<GetLoginStateUseCase>(FakeGetLoginStateUseCase());
+    Get.put<UserManager>(FakeUserManager());
+    Get.put<GetSurveysUseCase>(FakeGetSurveysUseCase());
+    Get.put<GetSurveyDetailUseCase>(FakeGetSurveyDetailUseCase());
+
+    testWidgets('When enter Home, it displays Survey carousel', (tester) async {
+      await tester.pumpWidget(SurveyApp());
+      await tester.pumpAndSettle();
+
+      await binding.watchPerformance(() async {
+        final pageView = find.byType(PageView);
+
+        expect(find.text('Today'), findsOneWidget);
+        expect(find.text('Title 1'), findsOneWidget);
+        expect(find.byType(FloatingActionButton), findsOneWidget);
+
+        // Move to the 2nd survey
+        await tester.fling(pageView, Offset(-200, 0), 2000);
+        await tester.pumpAndSettle();
+
+        expect(find.text('Title 2'), findsOneWidget);
+        expect(find.byType(FloatingActionButton), findsOneWidget);
+
+        // Move back to the 1st survey
+        await tester.fling(pageView, Offset(200, 0), 2000);
+        await tester.pumpAndSettle();
+
+        expect(find.text('Title 1'), findsOneWidget);
+        expect(find.byType(FloatingActionButton), findsOneWidget);
+      }, reportKey: 'scrolling_summary');
+    });
+
+    testWidgets('When choosing a survey, it displays detail', (tester) async {
```

Comment:
maybe we need more test cases. I.e. tapping on the avatar will show the Drawer, and tapping on Logout button will kick the user back to Login screen 💭

Reply:
that's a different scenario I didn't add the side drawer implementation yet 😂

Reply:
@sleepylee hmm, but you want to close the PR soon 😅

Reply:
![Screen Shot 2021-09-22 at 14 45 05](https://user-images.githubusercontent.com/13435717/134303399-ebb15b3c-d018-42b7-946b-401c015b9701.png)

@luongvo did I miss anything? 🤷  or did you guys assume all the nice-to-have features == must have?

Reply:
https://docs.google.com/document/u/1/d/e/2PACX-1vQERKjvXwKo3GpwRY4Qjc7U0afxjnPRcYEj6VK-scOEiZOYkolirtopUR8cZ2U65akiujAd0LKGp0ZS/pub

Reply:
oh, I see. So this issue is no longer valid. 🐤